packages/paste-core/components/slider/src/Slider.tsx

```diff
@@ -0,0 +1,160 @@
+   * and to preserve the underlying props object variable
+   */
+  const {
+    element = 'SLIDER',
+    minValue = 0,
+    maxValue = 100,
+    hideRangeLabels = false,
+    disabled: isDisabled = false,
+    id,
+    i18nMaxRangeLabel = 'Maximum value:',
+    i18nMinRangeLabel = 'Minimum value:',
+  } = props;
+
+  // Our API differs from the underlying library, so we need to remap props
+  const remappedProps = {
+    ...props,
+    id,
+    isDisabled,
+    // needed to silence react-aria a11y guardrails
+    'aria-labelledby': id,
+    numberFormatter: props.numberFormatter || DefaultNumberFormatter,
+  };
+
+  // These hooks manage the state of the slider
+  const state = useSliderState(remappedProps);
+  const {trackProps} = useSlider(remappedProps, state, trackRef);
+  const {thumbProps, inputProps, isDragging} = useSliderThumb(
+    {
+      index: 0,
+      trackRef,
+      inputRef,
+    },
+    state
+  );
+
+  // Used to determine the UI styling of the track and thumb
+  const uiStateProps = React.useMemo(() => {
+    return {
+      disabled: isDisabled,
+      error: props.hasError,
+      hovered,
+      focused,
+      dragging: isDragging,
+    };
+  }, [isDisabled, props.hasError, hovered, focused, isDragging]);
+
+  const onClickHandler = React.useCallback(() => {
+    inputRef?.current?.focus();
+  }, [inputRef]);
+
+  return (
+    <Box element={element}>
+      {/* Create a container for the optional min and max values */}
+      {!hideRangeLabels && (
+        <Box
+          element={`${element}_RANGE_LABELS`}
+          display="flex"
+          justifyContent="space-between"
+          fontSize="fontSize30"
+          lineHeight="lineHeight30"
+          fontWeight="fontWeightSemibold"
+          color="colorTextWeak"
+        >
+          <Box element={`${element}_RANGE_LABELS_MIN`}>
+            <ScreenReaderOnly>{i18nMinRangeLabel}</ScreenReaderOnly>
+            {remappedProps.numberFormatter.format(minValue)}
+          </Box>
+          <Box element={`${element}_RANGE_LABELS_MAX`}>
+            <ScreenReaderOnly>{i18nMaxRangeLabel}</ScreenReaderOnly>
+            {remappedProps.numberFormatter.format(maxValue)}
+          </Box>
+        </Box>
+      )}
+      {/* The track element holds the visible track line and the thumb */}
+      <Box
+        {...trackProps}
+        element={`${element}_TRACK_CONTAINER`}
+        ref={trackRef}
+        onMouseEnter={() => setHovered(true)}
+        onMouseLeave={() => setHovered(false)}
+        display="flex"
+        alignItems="center"
+        height="20px"
+        width="100%"
+        cursor={isDisabled ? 'not-allowed' : 'pointer'}
+        onClick={onClickHandler}
+      >
+        <StyledTrack
+          {...uiStateProps}
+          element={`${element}_TRACK`}
+          left={thumbProps.style?.left}
+          height="4px"
+          width="100%"
+          borderRadius="borderRadius20"
+          transition="background 150ms ease"
```

Comment:
issue: It seems like either the color transition isn't working on the track or appears to go at a much faster speed than the thumb.